### PR TITLE
Docugami DataLoader

### PR DESCRIPTION
**Pull request by eyurtsev**
### Adds a document loader for Docugami

Specifically:

1. Adds a data loader that talks to the [Docugami](http://docugami.com) API to download processed documents as semantic XML
2. Parses the semantic XML into chunks, with additional metadata capturing chunk semantics
3. Adds a detailed notebook showing how you can use additional metadata returned by Docugami for techniques like the [self-querying retriever](https://python.langchain.com/en/latest/modules/indexes/retrievers/examples/self_query_retriever.html)
4. Adds an integration test, and related documentation

Here is an example of a result that is not possible without the capabilities added by Docugami (from the notebook):

image

### CODE REVIEW

File: tests/unit_tests/document_loader/loaders/vendors/test_docugami.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""Test DocugamiLoader."""
```
Comment (eyurtsev) [inline tests/unit_tests/document_loader/loaders/vendors/test_docugami.py:1]: @tjaffri this was moved to unit tests

**Comment by eyurtsev** [comment]
cc @tjaffri PR has been merged -- made minor changes to resolve merge conflicts, some changes in type annotations and moved to unit tests folder

**Comment by tjaffri** [comment]
@eyurtsev thank you!

File: tests/unit_tests/document_loader/loaders/vendors/test_docugami.py (hunk 1 above)
Comment (tjaffri) [reply tests/unit_tests/document_loader/loaders/vendors/test_docugami.py:1]: Awesome thank you

**Pull request merged**